pkg/provider/kubernetes/client.go

```diff
@@ -120,17 +114,27 @@ func (c *Client) PushSecret(ctx context.Context, secret *v1.Secret, data esv1bet
 			if secret.Type != "" {
 				typ = secret.Type
 			}
-			return c.createSecret(ctx, value, typ, data)
+
+			return c.createSecret(ctx, secret, typ, data)
 		}
 		return getErr
 	}
-	// return gracefully if data is already in sync
-	if v, ok := extSecret.Data[data.GetProperty()]; ok && bytes.Equal(v, value) {
+
+	// the whole secret was pushed to the provider
+	if data.GetSecretKey() == "" {
+		if reflect.DeepEqual(extSecret.Data, secret.Data) {
+			return nil
+		}
+
+		return c.updateMap(ctx, extSecret, secret.Data)
+	}
```

Comment:
There's one more case that isn't covered/documented yet. We've covered so far:
* secretKey + remoteRef.property is defined
* secretKey + remoteRef.property is NOT defined
* not allowed: secretKey is defined + remoteRef.property is NOT defined

What's missing is:
* key is NOT defined, property is defined

That should marshal all the whole secret and stuff it into a property.

Reply:
Err... I thought that's exactly what I'm doing here: https://github.com/external-secrets/external-secrets/pull/2862/files#diff-8e6c015cd7592cc7f8958f9bf6a0097c4beed7819f421ec5d275c89000a95f9bR319 🤔

Reply:
I guess that's just when you initially create it, not when it's being updated

Reply:
Ah, right, you're talking about the update flow here.

Reply:
Yeah you're right. The update also has to consider marshaling it an pushing it into a property.

Reply:
@moolen Added it to update flow. :) nice one 👍